packages/react-ui/src/components/atoms/PasswordField.js

```diff
@@ -0,0 +1,35 @@
+import React, { forwardRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import { IconButton, InputAdornment, TextField } from '@mui/material';
+import { VisibilityOffOutlined, VisibilityOutlined } from '@mui/icons-material';
+
+const PasswordField = forwardRef(({ style, ...otherProps }, ref) => {
+  // forwardRef needed to be able to hold a reference, in this way it can be a child for some Mui components, like Tooltip
+  // https://mui.com/material-ui/guides/composition/#caveat-with-refs
+  const [showPassword, setShowPassword] = useState(false);
+  const handleClickShowPassword = () => setShowPassword(!showPassword);
+
+  return (
+    <TextField
+      {...otherProps}
+      ref={ref}
+      style={style}
+      type={showPassword ? 'text' : 'password'}
+      InputProps={{
+        endAdornment: (
+          <InputAdornment position='end'>
+            <IconButton onClick={handleClickShowPassword}>
+              {showPassword ? <VisibilityOutlined /> : <VisibilityOffOutlined />}
+            </IconButton>
+          </InputAdornment>
```

Comment:
Ok, I see the value here. At the beginning I thought we were just going to manage 'password' mode as hardcode. Obviously this makes more sense